netsim/cli/api.py

```diff
@@ -0,0 +1,384 @@
+  if root:
+    return root
+
+  return os.getcwd()
+
+
+def resolve_topology(payload: Dict[str, Any], workdir: str) -> str:
+  topology_url = (payload.get("topologyUrl") or "").strip()
+  if topology_url:
+    return topology_url
+  topology_path = (payload.get("topologyPath") or "").strip()
+  if not topology_path:
+    for candidate in ("netlab/topology.yml", "netlab/topology.yaml"):
+      if os.path.exists(os.path.join(workdir, candidate)):
+        topology_path = candidate
+        break
+  if not topology_path:
+    raise ValueError("topologyPath or topologyUrl required")
+  if os.path.isabs(topology_path):
+    return topology_path
+  return os.path.join(workdir, topology_path)
+
+
+def list_templates(template_dir: str) -> List[Dict[str, str]]:
+  results: List[Dict[str, str]] = []
+  if not template_dir:
+    return results
+  for root, dirs, files in os.walk(template_dir):
+    dirs[:] = [d for d in dirs if not d.startswith(".")]
+    for name in sorted(files):
+      if not name.endswith((".yml", ".yaml")):
+        continue
+      rel = os.path.relpath(os.path.join(root, name), template_dir)
+      results.append({"name": rel, "path": rel})
+    break
+  return results
+
+
+def run_netlab_action(payload: Dict[str, Any], log_fp: io.TextIOBase) -> None:
+  action = (payload.get("action") or "up").strip().lower()
+  workdir = workspace_dir(payload)
+  def _run_with_output(fn: Callable[[List[str]], None], args: List[str]) -> None:
+    out = io.StringIO()
+    with contextlib.redirect_stdout(out), contextlib.redirect_stderr(out):
+      fn(args)
+    log_fp.write(out.getvalue())
+
+  prev_cwd = os.getcwd()
+  try:
+    os.chdir(workdir)
+    if action == "up":
+      topology = resolve_topology(payload, workdir)
+      _run_with_output(netlab_up.run, [topology])
+    elif action == "create":
+      topology = resolve_topology(payload, workdir)
+      _run_with_output(netlab_create.run, [topology])
+    elif action == "down":
+      args: List[str] = []
+      if payload.get("cleanup"):
+        args.append("--cleanup")
+      _run_with_output(netlab_down.run, args)
+    elif action == "collect":
+      args = []
+      if payload.get("instance"):
+        args += ["--instance", str(payload.get("instance"))]
+      output_dir = payload.get("collectOutput")
+      if output_dir:
+        args += ["--output", output_dir]
+      tarball = payload.get("collectTar")
+      if tarball:
+        args += ["--tar", tarball]
+      if payload.get("collectCleanup"):
+        args.append("--cleanup")
+      _run_with_output(netlab_collect.run, args)
+    elif action == "status":
+      args = ["--all"]
+      if payload.get("instance"):
+        args = ["--instance", str(payload.get("instance"))]
+      _run_with_output(netlab_status.run, args)
+    else:
+      raise ValueError(f"unknown action {action}")
+  finally:
+    os.chdir(prev_cwd)
+
+
+def start_job(payload: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Can you start jobs in parallel? If that's the case, you need multilab plugin, or you have to serialize the jobs.